lib/vintage_net_wizard.ex

```diff
@@ -36,6 +36,31 @@ defmodule VintageNetWizard do
     end
   end
 
+  @doc """
+  Conditionally run the wizard if there is no configurations present
+
+  This function is the same `VintageNetWizard.run_wizard/1` however it will
+  first check if there are any configurations for the interface.
+
+  This is useful if you want a device to start the wizard only if there are no
+  configurations for the interface. When there are configurations found for the
+  interface this function returns `:configured` to let the consuming application
+  known that the wizard wasn't started because there are configurations found.
+
+  If you want more control on how to start the wizard or if you want to force
+  start the wizard you can use `VintageNetWizard.run_wizard/1` by itself.
+  """
+  @spec maybe_run_wizard([Endpoint.opt()]) :: :ok | :configured | {:error, String.t()}
+  def maybe_run_wizard(opts \\ []) do
```

Comment:
I'm wondering if there's a better name for the function?

```suggestion
  def run_if_unconfigured(opts \\ []) do
```